Feature/mail service tests

lucaserib opened the pull request:
Criando mocks e testes para rodar na entidade mail

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Adicionados mocks e suíte de testes abrangente para fluxos de e-mail (confirmação, recuperação, criação, alertas e envio genérico), cobrindo casos com/sem token e tratamento de erros, incluindo timeouts. Mensagens de teste atualizadas para português.
* **Bug Fixes**
  * Corrigido texto do assunto do e-mail de criação de empresa.
  * Corrigido campo de rede social (instagram).
* **Chores**
  * Atualizada composição de serviços e adicionado serviço de captura de e-mails em ambiente de desenvolvimento.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
feat: add unit tests for mail service

Commit:
fix: remove invalid savedJobs property from job mock

MikaelMelo1 commented on test/mocks/mail/company-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  companySize: 'SMALL_SIZE',
+  uf: 'SP',
+  otherSite: {
+    instagran: 'https://instagram.com/testcompany',
```
Pode ajustar o "instagran" para "instagram".

MikaelMelo1 commented on test/mocks/mail/company-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  password: 'hashedPassword123',
+  recoverPasswordToken: 'company-recover-token-456',
+  mailConfirm: false,
+  created_at: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função 
- new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/company-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  recoverPasswordToken: 'company-recover-token-456',
+  mailConfirm: false,
+  created_at: new Date('2023-01-01T00:00:00.000Z'),
+  updated_at: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função 
- new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/job-mail.mock.ts:
```diff
@@ -0,0 +1,50 @@
+  status: StatusEnum.ACTIVE,
+  content: null,
+  company_id: '123e4567-e89b-12d3-a456-426614174000',
+  createdAt: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função 
- new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/job-mail.mock.ts:
```diff
@@ -0,0 +1,50 @@
+  content: null,
+  company_id: '123e4567-e89b-12d3-a456-426614174000',
+  createdAt: new Date('2023-01-01T00:00:00.000Z'),
+  updatedAt: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função 
- new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/mail-options.mock.ts:
Ok! ✅

MikaelMelo1 commented on test/mocks/mail/company-mail.mock.ts:
Alguns Pontos.

MikaelMelo1 commented on test/mocks/mail/job-mail.mock.ts:
Alguns Pontos.

MikaelMelo1 commented on test/mocks/mail/user-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  mailConfirm: false,
+  type: 'USER',
+  policies: true,
+  created_at: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função

new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/user-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  type: 'USER',
+  policies: true,
+  created_at: new Date('2023-01-01T00:00:00.000Z'),
+  updated_at: new Date('2023-01-01T00:00:00.000Z'),
```
Pode deixar apenas com a função

new Date(), assim fica mais flexivel.

MikaelMelo1 commented on test/mocks/mail/user-mail.mock.ts:
Alguns pontos.

MikaelMelo1 commented on test/modules/mail/services/mail.service.spec.ts:
```diff
@@ -0,0 +1,342 @@
+      expect(mailerService.sendMail).toHaveBeenCalledTimes(1);
+      expect(mailerService.sendMail).toHaveBeenCalledWith({
+        to: company.email,
+        subject: 'Empresa criado!',
```
mude para "Empresa criada!"

MikaelMelo1 commented on test/modules/mail/services/mail.service.spec.ts:
Ok! ✅

MikaelMelo1 reviewed (changes requested):
Apenas alguns pontos para ajustes.

MikaelMelo1 reviewed (commented):
Beleza!

MikaelMelo1 reviewed (changes requested):
Alguns pontos.

MikaelMelo1 commented on test/mocks/mail/company-mail.mock.ts:
```diff
@@ -0,0 +1,32 @@
+  companyName: 'Test Company LTDA',
+  email: 'company@example.com',
+  cnpj: '12345678000199',
+  password: 'hashedPassword123',
```
Tem uma função simples pra gerar dinamicamente, da uma pesquisada. Assim, não fica visivel a senha, mesmo sendo em mock.

MikaelMelo1 reviewed (changes requested):
Da uma olhada.

Commit:
fix: corrigir imports do MailService e atualizar mensagens dos testes

- Corrigir imports relativos em mail.service.spec.ts e update-password-by-email.service.ts
- Adicionar MailService mock em update-password-by-email.service.spec.ts
- Atualizar mensagens esperadas de inglês para português nos testes
